### PR TITLE
Add models & login 

### LINKED ISSUE #2: Make Database
List of files
1. Events
2. User
3. User - event
4. User - Community
5. Community

### LINKED ISSUE #3: a user can make an account and login
1. login UI
2. API call for login 
3.sign up UI 
4. API call for sign up

### PR DESCRIPTION
**Pull request by robo360**
- Added a LoginActivity 
- Added models event and Community 

Note: LoginActivity has a login functionality with real api call to the parse server set up with fake data. 

Test Plan: models will be tested after building the mainActivity with a recyclerView showing events & the communities they belong to. The model User works as I tested it in the LoginActivity. 

Todo: 
- Add signup functionality 
- Create the rest of the models

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -35,5 +38,9 @@ dependencies {
     //parceler
     implementation 'org.parceler:parceler-api:1.1.12'
     annotationProcessor 'org.parceler:parceler:1.1.12'
+    //material design
+    implementation 'com.google.android.material:material:1.1.0'
+    implementation 'androidx.annotation:annotation:1.1.0'
+
 
 }
```
Comment (LynnHX) [inline app/build.gradle]:
```suggestion
}

```

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -10,14 +10,15 @@
...
                 <action android:name="android.intent.action.MAIN" />
 
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
         </activity>
+        <activity android:name=".activities.MainActivity">
+        </activity>
     </application>
 
 </manifest>
```
Comment (LynnHX) [inline app/src/main/AndroidManifest.xml]:
```suggestion
</manifest>

```

File: app/src/main/java/com/example/game/ParseApplication.java [hunk 3]
```diff
@@ -2,7 +2,12 @@
 
 import android.app.Application;
 
+import com.example.game.models.Community;
+import com.example.game.models.Event;
 import com.parse.Parse;
+import com.parse.ParseObject;
+
+import org.w3c.dom.Comment;
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/ParseApplication.java]:
```suggestion
```
Does not seem that you need this.

File: app/src/main/java/com/example/game/activities/LoginActivity.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import com.parse.ParseException;
+import com.parse.ParseUser;
+
+public class LoginActivity extends AppCompatActivity {
+    private ActivityLoginBinding binding;
+    private EditText etEmail;
+    private EditText etPassword;
+    private Button btnLogin;
+
+    public static final String  TAG = "LoginActivity";
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/activities/LoginActivity.java]:
```suggestion
    public static final String  TAG = "LoginActivity";
    
    private ActivityLoginBinding binding;
    private EditText etEmail;
    private EditText etPassword;
    private Button btnLogin;
```
Refer to java code style of parameter order.

File: app/src/main/java/com/example/game/activities/LoginActivity.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        btnLogin = binding.btnLogin;
+
+        btnLogin.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                //check whether the fields are not empty else login
+                String email = etEmail.getText().toString();
+                String password = etPassword.getText().toString();
+
+                if(email.isEmpty() || password.isEmpty()){
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/activities/LoginActivity.java]: nit:space
```suggestion
                if (email.isEmpty() || password.isEmpty()) {
```

File: app/src/main/java/com/example/game/activities/LoginActivity.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+                    Snackbar.make(btnLogin, "Email or password cannot be empty", BaseTransientBottomBar.LENGTH_SHORT);
+                }else{
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/activities/LoginActivity.java]: nit:space
```suggestion
                } else {
```

File: app/src/main/java/com/example/game/activities/LoginActivity.java [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+                    //Send Async request
+                    ParseUser.logInInBackground(email, password, new LogInCallback() {
+                        @Override
+                        public void done(ParseUser user, ParseException e) {
+                            if(e!= null){
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/activities/LoginActivity.java]: nit:space
```suggestion
                            if (e != null) {
```

File: app/src/main/java/com/example/game/activities/LoginActivity.java [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+                                Log.e(TAG, "Error while logging in" + e);
+                                Snackbar.make(btnLogin, "Incorrect username or password. Try again.", BaseTransientBottomBar.LENGTH_SHORT);
+                            }else{
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/activities/LoginActivity.java]: nit:space
```suggestion
                            } else {
```

File: app/src/main/java/com/example/game/activities/LoginActivity.java [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+    }
+
+    private void gotoMainActivity() {
+        //go to MainActivity
+        Intent i = new Intent(LoginActivity.this, MainActivity.class);
+        startActivity(i);
+        finish();
+    }
+
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/activities/LoginActivity.java]: nit:redundant new line
```suggestion
```

File: app/src/main/java/com/example/game/activities/LoginActivity.java [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+                            }else{
+                                Toast.makeText(LoginActivity.this, "Login Successful", Toast.LENGTH_SHORT).show();
+                                gotoMainActivity();
+                            }
+                        }
+                    });
+                }
+            }
+        });
+
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/activities/LoginActivity.java:71]: nit:redundant new line

```suggestion
```

File: app/src/main/java/com/example/game/activities/LoginActivity.java [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+    }
+
+    private void gotoMainActivity() {
+        //go to MainActivity
+        Intent i = new Intent(LoginActivity.this, MainActivity.class);
+        startActivity(i);
+        finish();
+    }
+
+}
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/activities/LoginActivity.java]: nit:new line end of file

```suggestion
}

```

File: app/src/main/res/drawable-v24/edit_text_border.xml [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+
+    <!-- Background Color -->
+    <solid android:color="#ffffff" />
+
+    <!-- Border Color -->
+    <stroke android:width="2dp" android:color="#039BE5" />
+
+    <!-- Round Corners -->
+    <corners android:radius="3dp" />
+</shape>
```
Comment (LynnHX) [inline app/src/main/res/drawable-v24/edit_text_border.xml]: nit:new line end of file
```suggestion
</shape>

```

File: app/src/main/res/layout/activity_login.xml [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".activities.LoginActivity">
+
+    <EditText
+        android:id="@+id/etMail"
+        android:layout_width="0dp"
+        android:layout_height="43dp"
+        android:layout_marginStart="16dp"
```
Comment (LynnHX) [inline app/src/main/res/layout/activity_login.xml]: nit: you could define certain standard margin/padding in dimens.xml. Reuse them and standardize the UI

File: app/src/main/res/layout/activity_login.xml [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+        android:layout_marginTop="16dp"
+        android:layout_marginEnd="16dp"
+        android:background="@drawable/edit_text_border"
+        android:ems="10"
+        android:gravity="center"
+        android:hint="password"
+        android:inputType="textMultiLine|textPassword"
+        android:textAlignment="center"
+        android:textColorHint="#FD777777"
```
Comment (LynnHX) [inline app/src/main/res/layout/activity_login.xml]: nit: you could define certain colors in colors.xml. Reuse them and standardize the UI

File: app/src/main/res/layout/activity_login.xml [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+        android:layout_height="wrap_content"
+        android:layout_marginTop="16dp"
+        android:text="New to Game.\n Sign up for an account.>>"
+        android:textAlignment="center"
+        android:textSize="16sp"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/btnLogin" />
+
+
```
Comment (LynnHX) [inline app/src/main/res/layout/activity_login.xml]: nit:redundant new line
```suggestion
```

File: app/src/main/res/layout/activity_login.xml [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (LynnHX) [inline app/src/main/res/layout/activity_login.xml]: nit:new line end of file

File: app/src/main/res/values/styles.xml [hunk 17]
```diff
@@ -1,10 +1,10 @@
...
         <!-- Customize your theme here. -->
-        <item name="colorPrimary">@color/colorPrimary</item>
-        <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
-        <item name="colorAccent">@color/colorAccent</item>
+        <item name="colorPrimary">#039BE5</item>
+        <item name="colorPrimaryDark">#1E88E5</item>
+        <item name="colorAccent">#00ACC1</item>
     </style>
 
 </resources>
```
Comment (LynnHX) [inline app/src/main/res/values/styles.xml]: nit: new line end of file

File: app/build.gradle [hunk 18]
```diff
@@ -35,5 +38,9 @@ dependencies {
     //parceler
     implementation 'org.parceler:parceler-api:1.1.12'
     annotationProcessor 'org.parceler:parceler:1.1.12'
+    //material design
+    implementation 'com.google.android.material:material:1.1.0'
+    implementation 'androidx.annotation:annotation:1.1.0'
+
```
Comment (LynnHX) [inline app/build.gradle:44]: nit:redundant new line
```suggestion
```

**Review by LynnHX — CHANGES_REQUESTED** [review]
Overall functionality lgtm. Let's revisit some styles.

File: app/src/main/java/com/example/game/activities/MainActivity.java [hunk 19]
```diff
@@ -1,9 +1,11 @@
-package com.example.game;
+package com.example.game.activities;
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/activities/MainActivity.java:1]: 👍 Good intention to structure files/folder

File: app/src/main/java/com/example/game/models/Community.java [hunk 20]
```diff
@@ -0,0 +1,63 @@
...
+
+import com.parse.Parse;
+import com.parse.ParseClassName;
+import com.parse.ParseFile;
+import com.parse.ParseObject;
+import com.parse.ParseUser;
+
+import org.parceler.Parcel;
+
+/*
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/models/Community.java:11]: 👍  Good comment

**Review by LynnHX — COMMENTED** [review]
nit: new line end of file.

Otherwise lgtm 😄

**Review by LynnHX — APPROVED** [review]

File: app/src/main/java/com/example/game/activities/LoginActivity.java [hunk 21]
```diff
@@ -19,12 +20,13 @@
 import com.parse.ParseUser;
 
 public class LoginActivity extends AppCompatActivity {
-    public static final String  TAG = "LoginActivity";
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/activities/LoginActivity.java]: Seems you reverted previous order change 😅

File: app/src/main/java/com/example/game/activities/SignupActivity.java [hunk 22]
```diff
@@ -0,0 +1,16 @@
...
+import com.example.game.R;
+
+public class SignupActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_signup);
+    }
+}
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/activities/SignupActivity.java]: nit:new line

File: app/src/main/res/layout/activity_signup.xml [hunk 23]
```diff
@@ -0,0 +1,111 @@
...
+        android:text="@string/loginMessage"
+        android:textAlignment="center"
+        android:textSize="16sp"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.497"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/btnLogin" />
+
+
+</androidx.constraintlayout.widget.ConstraintLayout>
```
Comment (LynnHX) [inline app/src/main/res/layout/activity_signup.xml]: nit:new line

File: app/src/main/res/values/strings.xml [hunk 24]
```diff
@@ -1,3 +1,11 @@
...
     <string name="app_name">Game</string>
+    <string name="setupAccount"><![CDATA[New to Game.\n Sign up for an account.>>]]></string>
+    <string name="e_mail">E-mail</string>
+    <string name="password">password</string>
+    <string name="login">Login</string>
+    <string name="repeat_password">Repeat Password</string>
+    <string name="name">Name</string>
+    <string name="singup">SingUp</string>
+    <string name="loginMessage">Have an account Login</string>
 </resources>
```
Comment (LynnHX) [inline app/src/main/res/values/strings.xml]: nit:new line

File: app/src/main/java/com/example/game/ParseApplication.java [hunk 25]
```diff
@@ -7,21 +7,20 @@
...
         ParseObject.registerSubclass(Community.class);
+
         // set applicationId, and server server based on the values in the Heroku settings.
         // clientKey is not needed unless explicitly configured
         // any network interceptors must be added with the Configuration Builder given this syntax
         Parse.initialize(new Parse.Configuration.Builder(this)
                 .applicationId("gameparse") // should correspond to APP_ID env variable
                 .server("https://gameparse.herokuapp.com/parse/").build());
     }
+
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/ParseApplication.java]: nit:redundant new line.

File: app/src/main/java/com/example/game/activities/LoginActivity.java [hunk 26]
```diff
@@ -20,14 +20,14 @@
 import com.parse.ParseUser;
 
 public class LoginActivity extends AppCompatActivity {
+    public static final String TAG = "LoginActivity";
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/activities/LoginActivity.java]: I think you probably only need this in this class so mark it as private

File: app/src/main/java/com/example/game/activities/MainActivity.java [hunk 27]
```diff
@@ -12,5 +12,8 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        //Run some tests of the models and the API
+
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/activities/MainActivity.java]: ???

File: app/src/main/java/com/example/game/models/Attendance.java [hunk 28]
```diff
@@ -0,0 +1,58 @@
...
+|updatedAt|	DateTime|	date when comment is last updated (default) field)|
+ */
+@Parcel(analyze = {Attendance.class})
+@ParseClassName("Attendance")
+public class Attendance extends ParseObject {
+    public static final String KEY_USER = "user";
+    public static final String KEY_EVENT = "event";
+    public static final String KEY_ATTEND_STATUS = "attendStatus";
+    public static final String KEY_LIKE_STATUS = "likeStatus";
+    public static final String KEY_CREATED_AT = "createdAt";
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/models/Attendance.java:29]: Seems you only access those from this class so mark as private

File: app/src/main/java/com/example/game/models/Comment.java [hunk 29]
```diff
@@ -0,0 +1,50 @@
...
+|comment |String | text of the comment written by the user|
+|createdAt|	DateTime|	date when the first interaction is created (default field)|
+|updatedAt|	DateTime|	date when comment is last updated (default) field)|
+ */
+@Parcel(analyze = {Comment.class})
+@ParseClassName("Comment")
+public class Comment extends ParseObject {
+    public static final String KEY_AUTHOR = "author";
+    public static final String KEY_EVENT = "event";
+    public static final String KEY_COMMENT = "comment";
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/models/Comment.java:25]: private

File: app/src/main/java/com/example/game/models/Subscription.java [hunk 30]
```diff
@@ -0,0 +1,70 @@
...
+ */
+@Parcel(analyze = {Subscription.class})
+@ParseClassName("Subscription")
+public class Subscription extends ParseObject {
+    public static final String KEY_USER = "user";
+    public static final String KEY_COMMUNITY = "community";
+    public static final String KEY_FOLLOW_STATUS = "followStatus";
+    public static final String KEY_EVENT_ATTEND = "eventAttend";
+    public static final String KEY_INTERACTION_COUNT = "interactionCount";
+    public static final String KEY_CREATED_AT = "createdAt";
```
Comment (LynnHX) [inline app/src/main/java/com/example/game/models/Subscription.java]: private

**Review by LynnHX — APPROVED** [review]
Overall lg.

File: app/src/main/java/com/example/game/models/Comment.java (hunk 29 above)
Comment (robo360) [reply app/src/main/java/com/example/game/models/Comment.java:25]: Makes sense!

File: app/src/main/java/com/example/game/activities/LoginActivity.java (hunk 26 above)
Comment (robo360) [reply app/src/main/java/com/example/game/activities/LoginActivity.java]: makes sense!

File: app/src/main/java/com/example/game/models/Attendance.java (hunk 28 above)
Comment (robo360) [reply app/src/main/java/com/example/game/models/Attendance.java:29]: I will access them later from one of my fragments.

Comment (robo360) [reply app/src/main/java/com/example/game/models/Attendance.java:29]: I will access them later in one of my fragments. I will need to access them without constructing an object.

File: app/src/main/java/com/example/game/models/Subscription.java (hunk 30 above)
Comment (robo360) [reply app/src/main/java/com/example/game/models/Subscription.java]: Will use them later!

**Pull request merged**